authorization-api-1_0.md

```diff
@@ -597,9 +600,186 @@ X-Request-ID: bfe9eb29-ab87-4ca3-be83-a1d5d8305720
+        "author",
+        "createDate",
+        "lastUpdated"
+    ]
+   },
+  "queries": ["delete", "read", "write"]
+}
+~~~
+{: #example-subject-search-request title="Example Subject Search Request"}
+
+### Subject Search Response {#subject-search-response}
+The success response to a Subject Search Request is a Subject Search Response. It is a HTTP response of type `application/json`. Its body is a JSON object that contains the following fields:
+
+iat:
+: REQUIRED. The issued at time in `integer` format, expressed as epoch milliseconds
+
+subject:
+: REQUIRED. The subject for which the response is being issued. The format of this field is as described in the Subjects section ({{subjects}})
+
+results:
+: REQUIRED. An array of Subject Query results as described below ({{subject-query-result}}).
+
+reasons:
+: OPTIONAL. An array of Reason Objects ({{reason-object}}) that describe the reason for each reason identified in the `decisions` field. This field is REQUIRED if there is at least one decision in the `decisions` field that specifies a `reason_ids` field. The content of the `reasons` field MUST provide details of every identifier in the `reason_ids` fields in the `decisions` array.
+
+nextPageToken:
+: OPTIONAL. A string that MAY be used in a Search Request to fetch the next set of responses.
+
+#### Subject Query Result {#subject-query-result}
+A Subject Query Result is JSON object combining a subject, a list of resource attribute names and an action. Given that a Subject Query result is expected to be the response to a Subject Search, only positive matches should be returned; i.e., only those subjects that match the search criteria (those subjects that are allowed to access the provided Resource Attributes). Any Subjects absent from the results do not have any access to the Resource. 
+A Subject Query Result has the following fields:
+
+actions:
+: OPTIONAL. An Array of the action for which the decision is provided. The format is as described in the Actions section ({{actions}}). The values in this list should match the values provided as queries in the Subject Search request.
+
+attributeNames:
+: OPTIONAL. An Array of attribute names of the resource for which the response applies. The attribute is provided only if attributes were part of the Subject search request. In that case, the attribute names must match those that are part of the request.
+
+subject:
+: REQUIRED. The subject for which the decision is provided. The format is as described in the Subjects section ({{subjects}}). 
+
+reason_ids:
+: OPTIONAL. An array of reason identifiers that indicate specific reasons why the resource query was denied
+
+The following is a non-normative example of a Subject Query Decision:
+
+~~~ json
+{
+    "actions": ["delete", "read", "write"],
+    "attributeNames": [
+        "author",
+        "createDate",
+        "lastUpdated"
+    ],
+    "subject": {
+        "id": "alex@3edges.com"
+    },
+    "reason_ids": [0,2,3]
```

Comment:
You probably need a "decision" string here that has either an "allow" or a "deny" value

Reply:
Actually, in the section above this, I stated: 
"Given that a Subject Query result is expected to be the response to a Subject Search, only positive matches should be returned; i.e., only those subjects that match the search criteria (those subjects that are allowed to access the provided Resource Attributes). Any Subjects absent from the results do not have any access to the Resource. "

If we also return the "Deny" subject, then we basically returning everybody, which is too much.

Reply:
Yeah, I agree we should return positive results only. I wonder if we should still have the reason codes then, because they may be useful to clients only if the decision is "deny". I can see some value to it, but I feel it will be OK to not return the reasons.

Reply:
Opened an issue about this, then closed it seeing this :) .

Anyway yes I tend to agree... ok will remove those from the results